shopinvader/services/sale.py

```diff
@@ -102,6 +125,18 @@ def _get_invoices(self, sale):
         domain_state = invoice_service._get_domain_state()
         return invoices.filtered_domain(domain_state)
 
+    def _cancel(self, sale, reset_to_cart=False):
+        for line in sale.order_line:
+            if line.qty_delivered > 0 or line.qty_invoiced > 0:
```

Comment:
You should use float_compare or float_is_zero

Reply:
IMHO the cancel behaviour could be conditionnable to something else (we agree this is the default).

So, I would have done a function to help people add their condtion for cancelable sales.

Reply:
@PierrickBrun can you fix it?